processor/transformprocessor/README.md

```diff
@@ -152,17 +161,55 @@ This is because contexts are nested: the efficiency comes because higher-level c
 ## Supported functions:
 
 Since the transform processor utilizes the OTTL's contexts for Traces, Metrics, and Logs, it is able to utilize functions that expect pdata in addition to any common functions. These common functions can be used for any signal.
-<!-- markdown-link-check-disable-next-line -->
+
 - [OTTL Functions](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/ottl/ottlfuncs)
 
 In addition to OTTL functions, the processor defines its own functions to help with transformations specific to this processor:
 
+**Common functions**
+- [drop](#drop)
+
 **Metrics only functions**
 - [convert_sum_to_gauge](#convert_sum_to_gauge)
 - [convert_gauge_to_sum](#convert_gauge_to_sum)
 - [convert_summary_count_val_to_sum](#convert_summary_count_val_to_sum)
 - [convert_summary_sum_val_to_sum](#convert_summary_sum_val_to_sum)
 
+## drop
+
+`drop()`
+
+Drops the specified [Context's](#contexts) telemetry from the collector.  Best used with a condition, otherwise all the Context's telemetry would be dropped.
+
+If a Metric ever has an empty DataPoints slice, the metric will also be dropped. 
+If a Scope ever has an empty Span/Metric/Log slice the scope will also be dropped. 
+If a Resource ever has an empty Scope slice the resource will also be dropped. 
+An empty SpanEvents slice will not cause a Span to be dropped.
+
+**Be very careful when dropping telemetry**.  Dropping telemetry can result in [unsound transformations or orphaned telemetry](#warnings).
+Drop does not attempt to reconcile any issues such as orphaned spans/logs or re-aggregation of metric datapoints.
+
+Examples:
+```yaml
+- context: datapoint
+  statements:
+    # drops any datapoint that has an attribute named "test" with a value of "pass".
+    - drop() where attributes["test"] == "pass"
+```
+```yaml
+- context: resource
+  statements:
+    # drops any resource (and therefore all its scopes and spans/metrics/logs) 
+    # that has an attribute named "test" with a value of "pass".
+    - drop() where attributes["test"] == "pass"
+```
+```yaml
+- context: spanevent
+  statements:
+    # drops any span event that has a name of "drop-me"
+    - drop() where name == "drop-me"
```

Comment:
Clarifying question: will this leave the actual span in place, just removing that one event ("drop-me")? Or will a span containing such an event be dropped entirely?

Reply:
So `name` is how you identify span events? what about any other thing that has a `name` on it?